tests/functional/common/vars-and-functions.sh

```diff
@@ -236,7 +236,8 @@ expect() {
     expected="$1"
     shift
     "$@" && res=0 || res="$?"
-    if [[ $res -ne $expected ]]; then
+    # also match "negative" codes, which wrap around to >127
+    if [[ $res -ne $expected && $res -ne $[256 + expected] ]]; then
```

Comment:
Do we want to just do modular arithmetic like `$(($res % 256))`?

Reply:
That would be clearer. I think I didn't believe we could.

Reply:
`%` preserves the sign, so it's not actually all that helpful. It just allows extra crazy shit like `expect 1025`. I don't think we need that.

Reply:
It's called modulo but doesn't actually convert the equivalence class into a canonical value. How pathetic.